### PR TITLE
Certificate update

### PR DESCRIPTION
**Pull request by poundbangbash**
Re-worked the existing certificate module to gather info about keychains found in the System.keychain as well as /etc/certificates.  Keychains deployed via config profiles would appear in the System.keychain and keeping tabs on their expiration is valuable.  Originally this module said it was for servers, this makes it a module available to clients as well.

The data gathered now includes the issuer of the certificate and the path of where the cert is stored.

Updated listing and client tab to include the actual expiration date, issuer, and location.
Created a widget and added to report to show certificates grouped by name.

### CODE REVIEW

**Comment by poundbangbash** [comment]
Wow. It was late. This is about getting data from certificates, not keychains as the original note references.

File: app/modules/certificate/locales/de.json [hunk 1]
```diff
@@ -1,7 +1,12 @@
 {
+    "cert_groups": "Certificates by Name"
```
Comment (bochoven) [inline app/modules/certificate/locales/de.json]: It looks like a comma is missing?

File: app/modules/certificate/locales/fr.json [hunk 2]
```diff
@@ -1,7 +1,12 @@
 {
+    "cert_groups": "Certificates by Name"
     "commonname": "Nom commun",
```
Comment (bochoven) [inline app/modules/certificate/locales/fr.json]: comma?

File: app/modules/certificate/migrations/001_certificate_add_columns.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        $dbh = $this->getdbh();
+
+        switch ($this->get_driver()) {
+            case 'sqlite':
+                $dbh->beginTransaction();
+
+                // Create temporary table
+                $sql = "CREATE TABLE certificate (
+								id INTEGER PRIMARY KEY, 
+								serial_number VARCHAR(255) UNIQUE,
```
Comment (bochoven) [inline app/modules/certificate/migrations/001_certificate_add_columns.php]: I don't think serial_number should be unique

**Comment by poundbangbash** [comment]
Thanks for pointing that out about the migration.  Not sure why it wasn't an issue in my testing.  I had multiple certs per SN.  Updated.

**Comment by bochoven** [comment]
I think we need more indexes:

```php
        $this->rs['cert_path'] = ''; // Path to certificate
        $this->rs['cert_cn'] = ''; // Common name
```

also need an index, I think

**Pull request merged**